### PR TITLE
Fix the problem that /debug/pprof/heap will wait 10s

### PR DESCRIPTION
**Pull request by CalvinNeo**
The problem is that according to https://github.com/pingcap/tidb-engine-ext/blob/50b8d2e0facdd8d7940ad6256a09945318cbaf6b/proxy_components/proxy_server/src/status_server/profile.rs#L104, the http API /debug/pprof/heap will use a ProfileGuard to sample at least the interval of time.

However, if we already enabled prof.active, it is no longer necessary to sleep this duration to sample. 

So we will remove this interval, and dump directly. And thus requires the user to explicitly turn on prof.active.

This behavior also correspond to the behavior that after we enabled continous profiling, which TiKV already did.

The change is that we totally use TiKV's idea that make `/heap` dump only the `prof.dump` file.

### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note

```

### CODE REVIEW

File: proxy_components/proxy_server/src/status_server/mod.rs [hunk 1]
```diff
@@ -301,7 +279,7 @@ where
                     .header("X-Content-Type-Options", "nosniff")
                     .header("Content-Disposition", "attachment; filename=\"profile\"")
                     .header("Content-Length", body.len());
-                response = if use_jeprof {
+                response = if use_jeprof && output_format == "--svg" {
```
Comment (JaySon-Huang) [inline proxy_components/proxy_server/src/status_server/mod.rs:282]: should it include `output_format == "--svg" || output_format == "--text" || output_format == "--collapsed"`?

Comment (JaySon-Huang) [reply proxy_components/proxy_server/src/status_server/mod.rs:282]: Oh, I get it. this if only set the HTTP response header.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: proxy_components/proxy_server/src/status_server/mod.rs (hunk 1 above)
Comment (CalvinNeo) [reply proxy_components/proxy_server/src/status_server/mod.rs:282]: No, --text and --raw will not output an svg

**Pull request merged**